src/python/pants/core/goals/update_build_files.py

```diff
@@ -380,11 +380,118 @@ def should_be_renamed(token: tokenize.TokenInfo) -> bool:
     )
 
 
+# ------------------------------------------------------------------------------------------
+# Rename deprecated field types fixer
+# ------------------------------------------------------------------------------------------
+
+
+class RenameDeprecatedFieldsRequest(DeprecationFixerRequest):
+    pass
+
+
+class RenamedFieldTypes(FrozenDict[Tuple[str, str], str]):
+    """Deprecated field type names for target to new names."""
+
+
+@rule
+def determine_renamed_field_types(
+    target_types: RegisteredTargetTypes, union_membership: UnionMembership
+) -> RenamedFieldTypes:
+    renamed_field_types = {}
+    for tgt in target_types.types:
+        for field_type in tgt.class_field_types(union_membership):
+            if field_type.deprecated_alias is None:
+                continue
+            renamed_field_types[(tgt.alias, field_type.deprecated_alias)] = field_type.alias
+            if tgt.deprecated_alias is not None:
+                renamed_field_types[
+                    (tgt.deprecated_alias, field_type.deprecated_alias)
+                ] = field_type.alias
+    return RenamedFieldTypes(renamed_field_types)
+
+
+@rule(desc="Check for deprecated field type names", level=LogLevel.DEBUG)
+def maybe_rename_deprecated_fields(
+    request: RenameDeprecatedFieldsRequest,
+    renamed_field_types: RenamedFieldTypes,
+) -> RewrittenBuildFile:
+    target = None
```

Comment:
Maybe call this `pants_target`. I wasn't sure until reading this for a few minutes if "target" meant some generic parsing term.